src/FantasyLand/Helpful/Tests/MonoidLawsTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FunctionalPHP\FantasyLand\Helpful\Tests;
+
+use Eris\Generator;
+use Eris\TestTrait;
+use FunctionalPHP\FantasyLand\Helpful\MonoidLaws;
+use FunctionalPHP\FantasyLand\Monoid;
+use FunctionalPHP\FantasyLand\Semigroup;
+
+class StringMonoid implements Monoid
+{
+    /**
+     * @var string
+     */
+    private $value;
+
+    public function __construct(string $value)
+    {
+        $this->value = $value;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function mempty()
+    {
+        return new self("");
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function concat(Semigroup $value): Semigroup
+    {
+        return new self($this->value . $value->value);
+    }
+}
+
+class MonoidLawsTest extends \PHPUnit_Framework_TestCase
+{
+    use TestTrait;
+
+    public function test_it_should_obay_monoid_laws()
+    {
+        $this->forAll(
+            Generator\string(),
+            Generator\string(),
+            Generator\string()
+        )->then(function (string $a, string $b, string $c) {
+            MonoidLaws::test(
```

Comment:
This is more a test of the `StringMonoid` rather than the "Law" itself.

I don't know if it is possible with PHPUnit, but it would be great to test failing cases to see if the `MonoidLaws` detect them corectly.

Reply:
This could however be the subject of a new issue so we can merge that quickly.

Reply:
Good point.